cogs/ratemyprofessor.py

```diff
@@ -0,0 +1,142 @@
+import os
+import sys
+import discord
+import yaml
+from discord.ext import commands
+import ratemyprofessor
+
+
+
+if not os.path.isfile("config.yaml"):
+    sys.exit("'config.yaml' not found! Please add it and try again.")
+else:
+    with open("config.yaml") as file:
+        config = yaml.load(file, Loader=yaml.FullLoader)
+
+
+class RateMyProfessor(commands.Cog, name="Rate my Professor"):
+    def __init__(self, bot):
+        self.bot = bot
+        self.profImages = {
+            "William Sverdlik": "https://www.emich.edu/computer-science/images/faculty/w-sverdlik.jpg",
+            "Zenia Bahorski": "https://www.emich.edu/computer-science/images/faculty/zbahorski.jpg",
+            "Andrii Kashliev": "https://www.emich.edu/computer-science/images/faculty/andreii-kashliev.jpg",
+            "Krish Narayanan": "https://www.emich.edu/computer-science/images/faculty/k-narayanan.jpg",
+            "Siyuan Jiang": "https://www.emich.edu/computer-science/images/faculty/s-jiang.jpg",
+        }
+
+    @commands.command(name="rmp")
+    async def rmp(self, context, *professorArgs):
+        """
+        [(Required) Professor name] Check out what RateMyProfessor has to say about a professor!
+        """
+        professorSearchTerm = " ".join(professorArgs)
+        EMU = ratemyprofessor.get_school_by_name("Eastern Michigan University")
+        prof = ratemyprofessor.get_professor_by_school_and_name(EMU, professorSearchTerm)
+
+        ratings =  sorted([rating for rating in prof.get_ratings() if rating.comment], key=lambda rating: (rating.rating, rating.date))
+        bestRating = ratings[-1]
+        worstRating = ratings[0]
+
+        profEmbed = self.buildProfEmbed(prof)
+        bestembed = self.buildRatingEmbed(discord.Embed(title=f"Best Rating for {prof.name}", color=config["success"]), bestRating)
+        worstembed = self.buildRatingEmbed(discord.Embed(title=f"Worst Rating for {prof.name}",color=config["success"]), worstRating)
+        
+        await context.send(embed=profEmbed)
+        await context.send(embed=bestembed)
+        await context.send(embed=worstembed)
+    
+    def buildRatingEmbed(self, embed, rating):
+        if rating.rating:
```

Comment:
A check is probably not required on all of these fields, but I'm adding them in case RMP ever makes any of these optional that used to be required so it never breaks the code.